### DISCUSSION TITLE
Cookie sessions lost on redeploy

### DISCUSSION THREAD

**Discussion by tslocke** [fastify/fastify#3936] [Q&A] [answered]
I'm using @fastify/session and fastify-cookie for auth. Every time I redeploy the app with new code, users get logged out. Any advice on how I can fix that? Here's the relevant part of my auth plugin:

```ts
import session from '@fastify/session'
import cookie from 'fastify-cookie'
import plugin from 'fastify-plugin'

export default plugin(async (app, options: any = {}) => {
  app.register(cookie)
  app.register(session, {
    cookieName: 'sessionId',
    secret: process.env.SESSION_SECRET!,
    cookie: { secure: false },
    saveUninitialized: false
  })
  ...
})
```

Thanks

**Comment by mcollina** [comment, marked as answer]
By default `@fastify/session` store things in a in-memory map. You need to configure a store. It should be compatible with https://github.com/expressjs/session#compatible-session-stores.

If you want a private session that is persisted in a cookie, use https://github.com/fastify/fastify-secure-session.

**Reply by tslocke** [reply]
Thanks - fastify-secure-session worked great.

Here's the code I ended up with, in case anyone else is interested

```ts
import plugin from 'fastify-plugin'
import '@fastify/secure-session'

export default plugin(async app => {
  app.register(require('@fastify/secure-session'), {
    secret: process.env.SESSION_SECRET,
    salt: process.env.SESSION_SALT,
    cookie: { path: '/', httpOnly: true }
  })
  ...
})
```
I had a type error if I did

```ts
import secureSession from '@fastify/secure-session'
...
app.register(secureSession, {...})
```

Which is why I have the slightly weird double import above, but no big deal.
